.travis.yml

```diff
@@ -9,33 +10,33 @@ jobs:
     - stage: "tests"
       name: "Run eslint"
       script: "npm install -g eslint@4; npm run eslint;"
-      node_js: "10"
-    - name: "TypeScript 2.9 generation and compilation tests"
-      node_js: "10"
+      node_js: "18"
+    - name: "TypeScript 4.9 generation and compilation tests"
+      node_js: "18"
       script:
-        - npm install -g typescript@2.9;
+        - npm install -g typescript@4.9;
```

Comment:
I'm not _completely_ convinced yet this is where we want to land but it's unquestionably an improvement over the current state of affairs.  We can always tweak it later if we decide we want to do something different.